### PR TITLE
GRW-697 - Show error modal if credit check fails

### PR DESCRIPTION
**Pull request by mehdielket**
## What?

Show error modal at checkout when backend returns a 403 with errorCode "MANUAL_REVIEW_REQUIRED"

## Why?

We need to alert users that they need to get in touch with our customer service teams

## How to test

* Need to get in touch with @fredriklagerblad to turn on the credit check on staging
* He or I can provide and NO SSN that generates this error

**Ticket(s): [GRW-697]**

## Screenshots / recordings 

https://user-images.githubusercontent.com/89857278/151935305-91301f9c-60d7-4ebc-a268-aaef1a475116.mov

[GRW-697]: https://hedvig.atlassian.net/browse/GRW-697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/pages/Offer/Checkout/index.tsx [hunk 1]
```diff
@@ -387,6 +399,11 @@ export const Checkout = ({
         return
       }
     } catch (e) {
```
Comment (robinandeer) [inline src/client/pages/Offer/Checkout/index.tsx:401]: consider renaming to `error`

File: src/client/pages/Offer/Checkout/index.tsx [hunk 2]
```diff
@@ -387,6 +399,11 @@ export const Checkout = ({
...
+      console.log(e.graphQLErrors, e.networkError)
+      if (isManualReviewRequired((e?.graphQLErrors || []) as GraphQLError[])) {
```
Comment (robinandeer) [inline src/client/pages/Offer/Checkout/index.tsx:403]: hmm just interesting - when is error `undefined`?

**Review by robinandeer — COMMENTED** [review]
Did we already turn off the credit check in staging? I used the same SSN as in your video but I could still sign.

Is it possible to add the "test instructions" that we need to contact IPC and which SSN:s to use in Notion?

Code looks good except for a few `console.log` statements :)

File: src/client/pages/Offer/Checkout/index.tsx (hunk 2 above)
Comment (mehdielket) [reply src/client/pages/Offer/Checkout/index.tsx:403]: Never really, it's just that error was not typed. The question mark is not needed

File: src/client/pages/Offer/Checkout/index.tsx [hunk 3]
```diff
@@ -441,60 +458,66 @@ export const Checkout = ({
...
-                  {textKeys.SIDEBAR_STARTDATE_CELL_LABEL()}
-                </StartDateLabel>
-                <StartDate quoteCartId={quoteCartId} offerData={offerData} />
-              </StartDateWrapper>
-              {isUpsellCardVisible && (
-                <UpsellCard
-                  quoteBundleVariants={quoteBundleVariants}
-                  selectedQuoteBundleVariant={selectedQuoteBundleVariant}
-                  onUpsellAccepted={onUpsellAccepted}
+      <>
```
Comment (guilhermespopolin) [inline src/client/pages/Offer/Checkout/index.tsx:461]: Wouldn't be the same as following? 🤔 

```
<>
  <OuterWrapper>
    ...
  </OuterWrapper>
  <Backdrop visibilityState={visibilityState} onClick={onClose} />
  <SignFailModal isVisible={isShowingFailModal} onClose={() => setIsShowingFailModal(false)} />
</>
```

**Review by guilhermespopolin — APPROVED** [review]

File: src/client/pages/Offer/Checkout/index.tsx (hunk 3 above)
Comment (mehdielket) [reply src/client/pages/Offer/Checkout/index.tsx:461]: I copied the markup from the old offer page. I can give your suggestion a try and see if it works

Comment (mehdielket) [reply src/client/pages/Offer/Checkout/index.tsx]: Fixed :)

**Review by guilhermespopolin — APPROVED** [review]
👍🏻

**Review by robinandeer — APPROVED** [review]

**Pull request merged**